1.8/usage/marathon/application-basics.md

```diff
@@ -0,0 +1,143 @@
+    ]
+}
+```
+
+What the example above does: before executing the `cmd`, download the resource `https://example.com/app/cool-script.sh` (via Mesos) and make it available in the service instance's sandbox. Verify that it had been downloaded by visiting the DC/OS web interface and clicking on an instance of `basic-1`, then on the **Files** tab. You should find `cool-script.sh` there.
+
+**Note:** As of Mesos v0.22 and above, the fetcher code does not make downloaded files executable by default. In the example above, `cmd` first makes the file executable.
+
+As already mentioned above, Marathon also [knows how to handle](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/mesos/TaskBuilder.scala) application resources that reside in archives. Currently, Marathon will (via Mesos and before executing the `cmd`) first attempt to unpack/extract resources with the following file extensions:
+
+* `.tgz`
+* `.tar.gz`
+* `.tbz2`
+* `.tar.bz2`
+* `.txz`
+* `.tar.xz`
+* `.zip`
+
+The following example shows you how this looks in practice. Assume you have an application executable in a zip file at `https://example.com/app.zip`. This zip file contains the script `cool-script.sh`, which you want to execute. Here's how:
+
+```json
+{
+    "id": "basic-2", 
+    "cmd": "app/cool-script.sh",
+    "cpus": 0.1,
+    "mem": 10.0,
+    "instances": 1,
+    "uris": [
+        "https://example.com/app.zip"
+    ]
+}
+```
+
+In contrast to the example `basic-1` we now have a `cmd` with the value `app/cool-script.sh`. When the zip file gets downloaded and extracted, a directory `app` according to the file name `app.zip` is created and the content of the zip file is extracted into it.
+
+You can specify more than one resource. For example, you could provide a git repository and some resources from a CDN as follows:
+
+```json
+{
+    "uris": [
+        "https://git.example.com/repo-app.zip", "https://cdn.example.net/my-file.jpg", "https://cdn.example.net/my-other-file.css"
+    ]
+}
+```
+
+A typical pattern in the development and deployment cycle is to have your automated build system place the app binary in a location that's downloadable via an URI. Marathon can download resources from a number of sources. Marathon supports the following [URI schemes](http://tools.ietf.org/html/rfc3986#section-3.1):
+
+* `file:`
+* `http:`
+* `https:`
+* `ftp:`
+* `ftps:`
+* `hdfs:`
+* `s3:`
+* `s3a:`
+* `s3n:`
+
+## A Simple Docker-based Application
+
+With Marathon it is straightforward to run applications that use Docker images.
+
+In the following example application definition, we will focus on a simple Docker app: a Python-based web server using the image [python:3](https://registry.hub.docker.com/_/python/). Inside the container, the web server runs on port `8080` (the value of `containerPort`). `hostPort` is set to `0` so that Marathon assigns a random port on the Mesos agent, which is mapped to port 8080 inside the container.
```

Comment:
Rewrite to explain that users are supposed to create an app named `basic-3.json`, and then deploy to DCOS using an API command. This is pretty hard to follow as-is.